src/internal_rules.jl

```diff
@@ -711,16 +716,17 @@ function EnzymeRules.reverse(
     ::Const{typeof(cholesky)},
     RT::Type,
     dfact,
-    A::Annotation{AT};
-    kwargs...) where {AT <: Array}
+    A::Annotation{<:Union{Matrix,LinearAlgebra.RealHermSym{<:Real,<:Matrix}}};
+    kwargs...)
 
     if !(RT <: Const) && !isa(A, Const)
         dAs = EnzymeRules.width(config) == 1 ? (A.dval,) : A.dval
         dfacts = EnzymeRules.width(config) == 1 ? (dfact,) : dfact
 
         for (dA, dfact) in zip(dAs, dfacts)
             if dA !== dfact.factors
```

Comment:
I'm wondering, should this check be changed as well?

Reply:
Yeah the check should make sure we're not adding to itself